tests/utils.test.ts

```diff
@@ -0,0 +1,271 @@
+    
+    const testResolveNodeModuleCachePath = async (modulePath: string, parseUrlPkg: any, cache: string) => {
+      try {
+        const moduleMetadata = await parseUrlPkg(modulePath);
+        const name = moduleMetadata?.pkg?.name;
+        const version = moduleMetadata?.pkg?.version;
+        const moduleFile = modulePath.split("/").reverse()[0] || "";
+        return `${cache}/${name}@${version}/${moduleFile}`;
+      } catch {
+        return undefined;
+      }
+    };
+    
+    const result = await testResolveNodeModuleCachePath(
+      'https://cdn.skypack.dev/lodash/index.js',
+      mockParseUrlPkg,
+      '.cache'
+    );
+    
+    assert.strictEqual(result, '.cache/lodash@4.17.21/index.js');
+  });
+
+  test('resolveNodeModuleCachePath handles missing metadata', async () => {
+    const mockParseUrlPkg = sinon.stub().resolves(null);
+    
+    const testResolveNodeModuleCachePath = async (modulePath: string, parseUrlPkg: any, cache: string) => {
+      try {
+        const moduleMetadata = await parseUrlPkg(modulePath);
+        const name = moduleMetadata?.pkg?.name;
+        const version = moduleMetadata?.pkg?.version;
+        if (!name || !version) return undefined;
+        const moduleFile = modulePath.split("/").reverse()[0] || "";
+        return `${cache}/${name}@${version}/${moduleFile}`;
+      } catch {
+        return undefined;
+      }
+    };
+    
+    const result = await testResolveNodeModuleCachePath(
+      'https://cdn.skypack.dev/unknown',
+      mockParseUrlPkg,
+      '.cache'
+    );
+    
+    assert.strictEqual(result, undefined);
+  });
+
+  test('resolveNodeModuleCachePath handles errors gracefully', async () => {
+    const mockParseUrlPkg = sinon.stub().rejects(new Error('Network error'));
+    
+    const testResolveNodeModuleCachePath = async (modulePath: string, parseUrlPkg: any, cache: string) => {
+      try {
+        const moduleMetadata = await parseUrlPkg(modulePath);
+        const name = moduleMetadata?.pkg?.name;
+        const version = moduleMetadata?.pkg?.version;
+        const moduleFile = modulePath.split("/").reverse()[0] || "";
+        return `${cache}/${name}@${version}/${moduleFile}`;
+      } catch {
+        return undefined;
+      }
+    };
+    
+    const result = await testResolveNodeModuleCachePath(
+      'https://cdn.skypack.dev/lodash',
+      mockParseUrlPkg,
+      '.cache'
+    );
+    
+    assert.strictEqual(result, undefined);
+  });
+
+  test('resolveParsedModulePath handles successful parsing', async () => {
+    const mockParseNodeModuleCachePath = sinon.stub().resolves('file:///cache/lodash@4.17.21/index.js');
+    
+    const testResolveParsedModulePath = async (modulePath: string, cachePath: string, parser: any) => {
+      try {
+        return await parser(modulePath, cachePath);
+      } catch {
+        return undefined;
+      }
+    };
+    
+    const result = await testResolveParsedModulePath(
+      'https://cdn.skypack.dev/lodash',
```

Comment:
I've personally been finding skypack a bit flakey recently - might be worth updating other CDN tests to unpkg / jsdelivr / esm.sh.